platforms/ios/example/Wysiwyg/Pills/WysiwygPermalinkReplacer.swift

```diff
@@ -15,18 +15,70 @@
 //
 
 import Foundation
-import HTMLParser
 import UIKit
+import WysiwygComposer
 
 final class WysiwygPermalinkReplacer: PermalinkReplacer {
-    func replacementForLink(_ link: String, text: String) -> NSAttributedString? {
+    func replacementForLink(_ url: String, text: String) -> NSAttributedString? {
         if #available(iOS 15.0, *),
-           link.starts(with: "https://matrix.to/#/"),
+           url.starts(with: "https://matrix.to/#/"),
            let attachment = WysiwygTextAttachment(displayName: text,
+                                                  url: url,
                                                   font: UIFont.preferredFont(forTextStyle: .body)) {
             return NSAttributedString(attachment: attachment)
         } else {
             return nil
         }
     }
+
+    func postProcessMarkdown(in attributedString: NSAttributedString) -> NSAttributedString {
+        // Create a regexp that detects markdown links.
+        let pattern = "\\[([^\\]]+)\\]\\(([^\\)\"\\s]+)(?:\\s+\"(.*)\")?\\)"
+        guard #available(iOS 15.0, *),
```

Comment:
What's the experience pre-iOS15, they see a link?

Reply:
Yeah, exactly.

Come to think maybe when we translate to markdown, we should display plain text display name rather (which means we will lose the link if toggling back to RTE). Right now toggling to plain after setting up a mention pre-iOS 15 would display the link mention in markdown.
Alternatively we could display both in plain text pre-iOS 15 which means people could just edit it like standard text, which is the behaviour of Element-iOS legacy composer.